### PR TITLE
lifecycle: Update RELEASES_SELECTOR

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
Remove old versions (keep 3 versions + release candid).
Update outdated version to use latest release of each minor.
Add missing releases.

**Special notes for your reviewer**:
Created this helper script, it will make this process less prone for mistakes
https://github.com/kubevirt/cluster-network-addons-operator/pull/1872

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by RamLavi** [comment]
Hey @oshoval thanks for the change,
This only changes the "short" lifecycle, that runs on every non-release PR
But we still have to update the "long" lifecycle, that runs over all the supported releases, and runs on every release PR. 
Please consider updating that as well.

**Comment by oshoval** [comment]
> Hey @oshoval thanks for the change, This only changes the "short" lifecycle, that runs on every non-release PR But we still have to update the "long" lifecycle, that runs over all the supported releases, and runs on every release PR. Please consider updating that as well.

Hi, I know and mentioned it as well on other PR today.
If you want please go ahead, atm I am focusing on short lifecyle.

File: test/releases/0.89.3.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+				ParentName: "kubemacpool-mac-controller-manager",
+				ParentKind: "Deployment",
+				Name:       "manager",
+				Image:      "quay.io/kubevirt/kubemacpool@sha256:ceaaed884bd2d9d4e26b5ebae20fe1023668076685a4eb101de2338395f5ebae",
+			},
+			{
+				ParentName: "kubemacpool-mac-controller-manager",
+				ParentKind: "Deployment",
+				Name:       "kube-rbac-proxy",
+				Image:      components.KubeRbacProxyImageDefault,
```
Comment (oshoval) [inline test/releases/0.89.3.go:52]: need to fix this, it should use the value hardcoded

**Comment by oshoval** [comment]
> @oshoval: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-e2e-cnao-multus-dynamic-networks-functests	[23c5c73](https://github.com/kubevirt/cluster-network-addons-operator/commit/23c5c735e74e854ed5434b0ccf59e3ddf1b2074a)	[link](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/1869/pull-e2e-cnao-multus-dynamic-networks-functests/1825104260623241216)	true	`/test pull-e2e-cnao-multus-dynamic-networks-functests`

https://github.com/kubevirt/cluster-network-addons-operator/issues/1871

**Comment by oshoval** [comment]
We should consider improving this flow, it can be semi automated
for now i just created quick and dirty hacks to add the missing files

**Comment by oshoval** [comment]
@RamLavi 
all lanes pass, do you want to review please ?

**Comment by oshoval** [comment]
/lgtm
/approve

according review i dont see something blocking
we can always fix in case any thing needed anyhow

**Comment by RamLavi** [comment]
I wanted to suggest that we wait until the GA date so that we could remove another release from the list.
But I see that you already forced merge.

**Comment by oshoval** [comment]
Thanks, but there is no reason to wait for that, we can always remove that once it comes.
I pro unblocking stuff, small PRs, unless there is a reason to create dependencies.
Moreover, i took here a window of 3 + RC + main, i think it is better than tight 3 + main.
We usually do that btw.

**Pull request merged**